src/Storages/ObjectStorage/DataLakes/HudiMetadata.cpp

```diff
@@ -94,7 +94,12 @@ HudiMetadata::HudiMetadata(ObjectStoragePtr object_storage_, ConfigurationObserv
 Strings HudiMetadata::getDataFiles() const
 {
     if (data_files.empty())
-        data_files = getDataFilesImpl();
+    {
+        for (const auto & data_file_name : getDataFilesImpl())
+        {
+            data_files.push_back(data_file_name);
+        }
+    }
```

Comment:
Why do we need this change?

Reply:
Yeap, thx fo noticing. These changes were relevant before you proposed solution how to avoid introducing new class DataInfo